### PR TITLE
new ad settings

### PR DESCRIPTION
**Pull request by alanrenmsft**
add two new AD related settings for CU5

### CODE REVIEW

File: extensions/resource-deployment/src/ui/deployClusterWizard/pages/clusterSettingsPage.ts [hunk 1]
```diff
@@ -186,6 +186,18 @@ export class ClusterSettingsPage extends WizardPageBase<DeployClusterWizard> {
...
+					label: localize('deployCluster.SubDomain', "Sub domain"),
+					required: false,
+					variableName: VariableNames.SubDomain_VariableName,
+					description: localize('deployCluster.SubDomainDescription', "A unique DNS subdomain to use for this BDC cluster. If not provided, cluster name will be used as the default value.")
+				}, {
+					type: FieldType.Text,
+					label: localize('deployCluster.AccountPrefix', "Account prefix"),
+					required: false,
+					variableName: VariableNames.AccountPrefix_VariableName,
+					description: localize('deployCluster.AccountPrefixDescription', "A unique prefix for AD accounts BDC cluster will generate,When not provided, subdomain name will be used as the default value. When subdomain is not provided, cluster name will be used as the subdomain name.")
```
Comment (Charles-Gagnon) [inline extensions/resource-deployment/src/ui/deployClusterWizard/pages/clusterSettingsPage.ts:200]: Should the comma between generate,When be a period and a space instead?

Comment (Charles-Gagnon) [inline extensions/resource-deployment/src/ui/deployClusterWizard/pages/clusterSettingsPage.ts:200]: `When a subdomain is not provided, the cluster name...`

Comment (Charles-Gagnon) [inline extensions/resource-deployment/src/ui/deployClusterWizard/pages/clusterSettingsPage.ts:200]: `BDC cluster` == `Big Data Clusters cluster`? Seems like normally we use the full product name `Big Data Cluster` when referring to an actual cluster

File: extensions/resource-deployment/src/ui/deployClusterWizard/pages/clusterSettingsPage.ts [hunk 2]
```diff
@@ -186,6 +186,18 @@ export class ClusterSettingsPage extends WizardPageBase<DeployClusterWizard> {
 					variableName: VariableNames.AppReaders_VariableName,
 					placeHolder: localize('deployCluster.AppReadersPlaceHolder', "Use comma to separate the values."),
 					description: localize('deployCluster.AppReadersDescription', "The Active Directory users or groups of app readers. Use comma as separator them if there are multiple users/groups.")
+				}, {
+					type: FieldType.Text,
+					label: localize('deployCluster.SubDomain', "Sub domain"),
+					required: false,
+					variableName: VariableNames.SubDomain_VariableName,
+					description: localize('deployCluster.SubDomainDescription', "A unique DNS subdomain to use for this BDC cluster. If not provided, cluster name will be used as the default value.")
```
Comment (Charles-Gagnon) [inline extensions/resource-deployment/src/ui/deployClusterWizard/pages/clusterSettingsPage.ts:194]: `If not provided, the cluster name`

File: extensions/resource-deployment/src/ui/deployClusterWizard/pages/summaryPage.ts [hunk 3]
```diff
@@ -162,6 +162,21 @@ export class SummaryPage extends WizardPageBase<DeployClusterWizard> {
 						labelFontWeight: FontWeight.Bold
 					}]
 			});
+			clusterSectionInfo.rows!.push({
+				fields: [
+					{
+						type: FieldType.ReadonlyText,
+						label: localize('deployCluster.SubDomain', "Sub domain"),
```
Comment (Charles-Gagnon) [inline extensions/resource-deployment/src/ui/deployClusterWizard/pages/summaryPage.ts:169]: Subdomain

File: extensions/resource-deployment/src/ui/deployClusterWizard/constants.ts [hunk 4]
```diff
@@ -17,8 +17,10 @@ export const DomainDNSName_VariableName = 'AZDATA_NB_VAR_BDC_AD_DOMAIN_DNS_NAME'
 export const Realm_VariableName = 'AZDATA_NB_VAR_BDC_AD_REALM';
 export const DomainServiceAccountUserName_VariableName = 'AZDATA_NB_VAR_BDC_AD_DOMAIN_SVC_USERNAME';
 export const DomainServiceAccountPassword_VariableName = 'AZDATA_NB_VAR_BDC_AD_DOMAIN_SVC_PASSWORD';
-export const AppOwners_VariableName = 'AZDATA_NB_VAR_AD_BDC_APP_OWNERS';
-export const AppReaders_VariableName = 'AZDATA_NB_VAR_AD_BDC_APP_READERS';
+export const AppOwners_VariableName = 'AZDATA_NB_VAR_BDC_AD_APP_OWNERS';
+export const AppReaders_VariableName = 'AZDATA_NB_VAR_AD_BDC_AD_APP_READERS';
+export const SubDomain_VariableName = 'AZDATA_NB_VAR_BDC_AD_SUBDOMAIN';
```
Comment (Charles-Gagnon) [inline extensions/resource-deployment/src/ui/deployClusterWizard/constants.ts:22]: Shouldn't you be updating the required azdata version as well?

Comment (alanrenmsft) [reply extensions/resource-deployment/src/ui/deployClusterWizard/constants.ts:22]: i am following up separately for the version update. will have a separate PR for that.

**Pull request merged**